src/Tools/dotnet-monitor/CollectionRules/Actions/CollectDumpAction.cs

```diff
@@ -3,19 +3,80 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Monitoring.WebApi.Controllers;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Exceptions;
 using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Extensions.DependencyInjection;
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions
 {
-    internal sealed class CollectDumpAction :
-        ICollectionRuleAction<CollectDumpOptions>
+    internal sealed class CollectDumpAction : ICollectionRuleAction<CollectDumpOptions>
     {
-        public Task<CollectionRuleActionResult> ExecuteAsync(CollectDumpOptions options, IEndpointInfo endpointInfo, CancellationToken token)
+        private readonly IDiagnosticServices _diagnosticServices;
+        private readonly IServiceProvider _serviceProvider;
+
+        public CollectDumpAction(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+            _diagnosticServices = serviceProvider.GetRequiredService<IDiagnosticServices>();
+        }
+
+        public async Task<CollectionRuleActionResult> ExecuteAsync(CollectDumpOptions options, IEndpointInfo endpointInfo, CancellationToken token)
+        {
+            DumpType dumpType = options.Type.GetValueOrDefault(CollectDumpOptionsDefaults.Type);
+            string egressProvider = options.Egress;
+
+            IProcessInfo processInfo = await _diagnosticServices.GetProcessAsync(new ProcessKey(endpointInfo.ProcessId), token);
+
+            string dumpFileName = DiagController.GenerateDumpFileName();
```

Comment:
Avoid this kind of coupling. Consider a separate utility class for generating the name rather than making Dump actions depend on WebServer controllers.